src/frontend/src/pages/AdminToolsPage/OnboardingConfig/Checklists/CreateChecklistModal.tsx

```diff
@@ -0,0 +1,328 @@
+  teamId?: string;
+  teamTypeId?: string;
+}
+
+interface ChecklistFormValues {
+  name: string;
+  descriptions: { name: string }[];
+  subtasks: ChecklistPreview[];
+}
+
+const CreateChecklistModal = ({ open, handleClose, teamId, teamTypeId }: CreateChecklistModalProps) => {
+  const theme = useTheme();
+  const toast = useToast();
+  const { mutateAsync: createChecklist, isLoading, isError, error } = useCreateChecklist();
+
+  const handleFormSubmit = async (data: ChecklistCreateArgs) => {
+    try {
+      const response = await createChecklist(data);
+      return response;
+    } catch (err) {
+      toast.error('Failed to create checklist');
+      throw err;
+    }
+  };
+
+  const [subtasks, setSubtasks] = useState<ChecklistPreview[]>([
+    { name: '', isOptional: false, dateCreated: new Date(), checklistId: '' }
+  ]);
+
+  const schema = yup.object().shape({
+    name: yup.string().required('Name is Required'),
+    descriptions: yup
+      .array()
+      .of(
+        yup.object().shape({
+          name: yup.string().required('Description is Required')
+        })
+      )
+      .min(1, 'At least one description is required'),
+    subtasks: yup.array().of(
+      yup.object().shape({
+        name: yup.string().required('Subtask Name is Required'),
+        isOptional: yup.boolean().required('Is Optional is Required')
+      })
+    )
+  });
+
+  const {
+    handleSubmit,
+    control,
+    reset,
+    formState: { errors }
+  } = useForm<ChecklistFormValues>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      name: '',
+      descriptions: [{ name: '' }],
+      subtasks: []
+    }
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'descriptions'
+  });
+
+  if (isError) return <ErrorPage message={error?.message} />;
+  if (isLoading) return <LoadingIndicator />;
+
+  const onFormSubmit = async (data: ChecklistCreateArgs) => {
+    try {
+      const formattedData = {
+        ...data,
+        teamId,
+        teamTypeId,
+        descriptions: (data.descriptions as unknown as { name: string }[]).map((desc) => desc.name)
+      };
+
+      const parentChecklist = await handleFormSubmit(formattedData);
+
+      // Handle subtasks
+      const filteredSubtasks = subtasks.filter((subtask) => subtask.name !== '');
+      await Promise.all(
+        filteredSubtasks.map((subtask) =>
+          handleFormSubmit({
+            name: subtask.name,
+            descriptions: [],
+            teamId,
+            teamTypeId,
+            parentChecklistId: parentChecklist.checklistId,
+            isOptional: subtask.isOptional
+          })
+        )
+      );
+
+      handleClose();
+    } catch (error) {
+      toast.error('Failed to create checklist');
+      console.error('Error in onFormSubmit:', error);
+    }
+  };
+
+  const addSubtask = () => {
+    setSubtasks([...subtasks, { name: '', isOptional: false, dateCreated: new Date(), checklistId: '' }]);
```

Comment:
why do we need to be passing in dateCreated? I would assume the createChecklistargs does not take in a dateCreated